src/plugins/vis_type_table/public/plugin.ts

```diff
@@ -34,8 +34,7 @@ export interface TablePluginStartDependencies {
 
 /** @internal */
 export class TableVisPlugin
-  implements
-    Plugin<Promise<void>, void, TablePluginSetupDependencies, TablePluginStartDependencies> {
+  implements AsyncPlugin<void, void, TablePluginSetupDependencies, TablePluginStartDependencies> {
```

Comment:
@elastic/kibana-app I couldn't migrate `vis_type_table` because of the conditional lazy imports performed from the plugin

https://github.com/elastic/kibana/blob/36e4e4a9f2bf90401f555d1220d92a27a7068088/src/plugins/vis_type_table/public/plugin.ts#L50-L56

However, I agree that conditionally loading the vis types depending on the config value seems like a good pattern, but I'm unsure how we will be able to handle that in synchronous plugins. I'll ping you on the follow-up to discuss about that.

cc @elastic/kibana-core